### DISCUSSION TITLE
How to check for go.mod

### DISCUSSION THREAD

**Discussion by caibirdme** [golangci/golangci-lint#4972] [Q&A] [answered]
Seems like the `go/analysis` just check for `xx.go` file, how to write a linter for checking `go.mod`(like force people to use some specific version of a package)?

I tried but a little confusing.

1. If I want to report some warnings, how should I call the `pass.Reportf`? It requires a `token.Pos` but I don't have a Pos
2. If I just set the `token.Pos` as `pass.Files[0].Pos()`, it reports a *invalid_issue* such as:
```
WARN [runner/invalid_issue] no file path for the issue: probably a bug inside the linter "gomodcheck": &result.Issue{FromLinter:"gomodcheck", Text:"dependency xxxx must be at least version 0.19.0, but found 0.18.4", Severity:"", SourceLines:[]string(nil), Replacement:(*result.Replacement)(nil), Pkg:(*packages.Package)(0xc000002000), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"", Offset:0, Line:0, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""} 
```
How can I report a *valid* issue?

**Comment by ldez** [comment, marked as answer]
Hello,

`go/analysis` cannot be used to analyze `go.mod`.

FYI, we already have linters dedicated to `go.mod`:
- https://golangci-lint.run/usage/linters/#gomoddirectives
- https://golangci-lint.run/usage/linters/#gomodguard

**Reply by caibirdme** [reply]
Thx very much
